core/build.gradle.kts

```diff
@@ -187,6 +191,40 @@ ide {
     generatedSource(javaCCTest, "test")
 }
 
+fun JavaCompile.configureAnnotationSet(sourceSet: SourceSet) {
+    source = sourceSet.java
+    classpath = sourceSet.compileClasspath
+    destinationDirectory.set(options.generatedSourceOutputDirectory)
+    options.compilerArgs.add("-proc:only")
+    org.gradle.api.plugins.internal.JvmPluginsHelper.configureAnnotationProcessorPath(sourceSet, sourceSet.java, options, project)
+    // only if we aren't running compileJava, since doing twice fails.
+    onlyIf { project.gradle.taskGraph.hasTask(sourceSet.getCompileTaskName("java")) }
```

Comment:
I belive `hasTask` condition should be reversed.

Reply:
Nice catch. 

The jvm bug is apparently only triggered in the test kotlin/java combination (which is why the build still worked here). 

Fixed now.